packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -372,6 +370,137 @@ export class CloudFormationDeployments {
     return stack.exists;
   }
 
+  private async readCurrentStackTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, stackName: string, stackSdk: ISDK) : Promise<Template> {
+    // if `stackName !== stackArtifact.stackName`, then `stackArtifact` is an ancestor to a nested stack with name `stackName`.
+    debug(`Reading existing template for stack ${stackArtifact.displayName}.`);
+    const cfn = stackSdk.cloudFormation();
+    const stack = await CloudFormationStack.lookup(cfn, stackName);
+
+    return stack.template();
+  }
+
+  private async prepareSdkForLookup(stackArtifact: cxapi.CloudFormationStackArtifact): Promise<ISDK> {
```

Comment:
I have to say, the name of this method doesn't make much sense to me 😛. What "lookup" are we preparing for here?

I think what you need here is something like `prepareSdkWithLookupOrDeployRole()`, or something like that.